code/datums/mind.dm

```diff
@@ -1488,29 +1488,25 @@
  * Create and/or add the `datum_type_or_instance` antag datum to the src mind.
  *
  * Arguments:
- * * datum_type - an antag datum typepath or instance
+ * * antag_datum - an antag datum typepath or instance. If it's a typepath, it will create a new antag datum
  * * datum/team/team - the antag team that the src mind should join, if any
  */
-/datum/mind/proc/add_antag_datum(datum_type_or_instance, datum/team/team = null)
-	var/datum/antagonist/A
-	if(!ispath(datum_type_or_instance))
-		A = datum_type_or_instance
-		if(!istype(A))
-			return
-	else
-		A = new datum_type_or_instance()
-	if(!A.can_be_owned(src))
-		qdel(A)
+/datum/mind/proc/add_antag_datum(datum/antagonist/antag_datum, datum/team/team)
+	if(!istype(antag_datum))
+		antag_datum = new()
+
+	if(!antag_datum.can_be_owned(src))
+		qdel(antag_datum)
```

Comment:
I don't like this, this could potentially create a `/datum/antagonist`, I think the old version is better

Reply:
Sounds good, I'll revert it

Reply:
Reverted in the next commit